src/redux/stock/reducer.js

```diff
@@ -0,0 +1,29 @@
+import { FETCH_STOCK_DETAILS_START, FETCH_STOCK_DETAILS_SUCCESS, FETCH_STOCK_DETAILS_FAIL } from './actionType';
+
+const initState = {
+  loading: false,
+  details: undefined,
+  error: undefined,
+};
+
+export default (state = initState, action) => {
+  switch (action.type) {
+    case FETCH_STOCK_DETAILS_START:
+      return {
+        ...initState,
+        loading: true,
+      };
+    case FETCH_STOCK_DETAILS_SUCCESS:
+      return {
+        ...initState,
+        details: action.payload,
```

Comment:
Now it is fine, but later in well-separated applications, we should keep in mind modularity and flexibility. Eg if later on, we replace the third party API then we might get different field names, so we will need to change the component. This is not a good practice, the component is totally independent from the data source. Actually here or in the action we should map the response fields to our own field names, so in case of api change we just need to change the code only in one file - this is why i would prefer the mapping in the action file. 
(Of course you dont need to do this now, it is just a future suggestion) - more info in the clean architecture book

Reply:
Will do this when we combine APIs for the detailed fields